src/phase_3_ransac_filtering/ransac_filter.py

```diff
@@ -0,0 +1,190 @@
+                         img1_nodata=None,
+                         img2_filename=None,
+                         img2_nodata=None):
+    '''ASIFT_RANSAC_filter::filter_ASIFT_matches(matching_keypoints_filename,
+                                                 output_good_points_only = True,
+                                                 verbose = False,
+                                                 ransac_reprojection_threshold=3)
+        Parameters:
+            * matching_keypoints_filename (req'd): the file path of the "data_matches.csv" file produced by the ASIFT executable.
+            * output_good_points_only (opt'l): boolean, default True
+                                               if True: return a 2 lists of matching cv2::KeyPoint matches, bad points omitted.
+                                               if False: return the Homography transformatio matrix, keypoints1 (all), keypoints2 (all), and a boolean mask of the selected keypoints in each array.
+            * verbose (opt'l): boolean, default False
+                               if True: print a status statement when the function completes.
+                               if False: stay silent, just return the damned matches.
+            * ransac_reprojection_threshold (opt'l): number 1-10, default 3. The allowed cutoff threshold for accepting/rejecting transformed matching keypoints as "good" matches.
+    '''
+
+    # 1. Open the "data_matches.csv" file, digest the points into matches and keypoints manually.
+    keys1, keys2, matches = ASIFT_Matches(matching_keypoints_filename).return_cv2_keypoints()
+
+    keys1_ndarray = numpy.asarray([keys1[i].pt for i in range(len(keys1))])
+    keys2_ndarray = numpy.asarray([keys2[i].pt for i in range(len(keys2))])
+
+    # 1.5 -- eliminate values that are on or adjacent to nodata values.
+    if eliminate_nodata_matches:
+        if img1_filename != None:
+            img1_nodata_mask = find_nodata_matches(keys1_ndarray, img1_filename, img1_nodata)
+        else:
+            img1_nodata_mask = numpy.zeros((keys1_ndarray.shape[0],),dtype=numpy.bool)
+
+        if img2_filename != None:
+            img2_nodata_mask = find_nodata_matches(keys2_ndarray, img2_filename, img2_nodata)
+        else:
+            img2_nodata_mask = numpy.zeros((keys2_ndarray.shape[0],),dtype=numpy.bool)
+
+        bad_matches_mask = img1_nodata_mask | img2_nodata_mask
+        assert len(bad_matches_mask) == len(keys1) == len(keys2)
+
+    else:
+        bad_matches_mask = numpy.zeros((keys1_ndarray.shape[0],),dtype=numpy.bool)
+
+
+    # 2. Use the cv2.findHomography call, and tweak the parameters to get a good filter.
+    # See reference info: https://docs.opencv.org/3.0-beta/modules/calib3d/doc/camera_calibration_and_3d_reconstruction.html?highlight=findhomography#findhomography
+
+    H, mask = cv2.findHomography(keys1_ndarray, keys2_ndarray, method=cv2.RANSAC, ransacReprojThreshold=ransac_threshold, maxIters=2000)
```

Comment:
@mmacferrin , 

in this part, I am getting this error , `H, mask = cv2.findHomography(keys1_ndarray, keys2_ndarray, method=cv2.RANSAC, ransacReprojThreshold=ransac_threshold, maxIters=2000)
TypeError: 'maxIters' is an invalid keyword argument for this function` 

so i fixed it 2 days ago and created a pull request under the branch `fix_ransac` by using the following 
 `H, mask = cv2.findHomography(keys1_ndarray, keys2_ndarray, cv2.RANSAC, ransac_threshold, 2000)` .